File: go/solutions/0392-is-subsequence.go
package solutions

func IsSubsequence(s string, t string) bool {
	j := len(s) - 1
	for i := len(t) - 1; i >= 0 && j >= 0; i-- {
		if t[i] == s[j] {
			j--
			continue
		}
	}

	return j < 0
}

File: js/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    var j = s.length-1;
    for (var i = t.length-1; i >= 0 && j >= 0; i--) {
        if (t.charCodeAt(i) === s.charCodeAt(j)) {
            j--;
        }
    }

    return j < 0;
};

module.exports = { isSubsequence }